File: app/src/main/java/com/example/bluetooth_le_gatt_sborka/Activity3.java
package com.example.bluetooth_le_gatt_sborka;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.bluetooth_le_gatt_sborka.support.AcResult;

public class Activity3 extends AppCompatActivity implements View.OnClickListener {
    public static final String EXTRA_ACRESULT = "acresult";

    public static final String ACTION_FINISH_UPDATING = "ACTION_ASYNC_FINISH_UPDATING";
    private static final String TAG = "A3";
    private Button buttonBack;
    private AcResult mAcResult;
    private TextView
            viewTime,
            viewValue,
            viewDate,
            viewDeviceType,
            viewDevice,
            viewDriver;

    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setResult(RESULT_CANCELED);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.result);
        Intent intent = getIntent();
        mAcResult = (AcResult) intent.getSerializableExtra(EXTRA_ACRESULT);
        initializeView(mAcResult);
        buttonBack = findViewById(R.id.button_back);
        buttonBack.setOnClickListener(this);
    }

    private void initializeView(AcResult acResult) {
        viewDate = findViewById(R.id.ac_date);
        viewDate.setText(acResult.getAcDate());
        viewTime = findViewById(R.id.ac_time);
        viewTime.setText(acResult.getAcTime());
        viewValue = findViewById(R.id.ac_value);
        viewValue.setText(acResult.getAcValue());

        float value = Float.parseFloat(viewValue.getText().toString().substring(0, 5));
        if (value >= 0.16f) {
            viewValue.setTextColor(getResources().getColor(R.color.red));
        } else {
            viewValue.setTextColor(getResources().getColor(R.color.green));
        }

        viewDriver = findViewById(R.id.driver);
        // if(isEnabled){
        //     viewDriver.setText(acResult.getUserText() + "(Водитель был удален!)");
        //     viewDriver.setBackgroundColor(res.getColor(R.color.red));
        // }
        // else
        viewDriver.setText("Z");//acResult.getUserText());

        viewDeviceType = findViewById(R.id.deviceType);
        switch (acResult.getTestType()) {
            case "alco_test":
                viewDeviceType.setText(R.string.caption_device);
                break;
            case "pres_test":
                viewDeviceType.setText(R.string.caption_device2);
                break;
            case "temp_test":
                viewDeviceType.setText(R.string.caption_device3);
                break;
            default:
                break;
        }
        viewDevice = findViewById(R.id.device_name);
        viewDevice.setText(acResult.getDeviceName());
    }

    public void onClick(View v) {
        int id = v.getId();
        if (id == R.id.button_back) {
            setResult(RESULT_OK);
            finish();
        }
    }
}

File: app/src/main/java/com/example/bluetooth_le_gatt_sborka/Activity3.kt
package com.example.bluetooth_le_gatt_sborka

import android.os.Bundle
import android.view.View
import android.view.Window
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.example.bluetooth_le_gatt_sborka.support.AcResult

class Activity3 : AppCompatActivity(), View.OnClickListener {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.result)
        val mAcResult = intent.getSerializableExtra(EXTRA_ACRESULT) as AcResult
        initializeView(mAcResult)
        findViewById<Button>(R.id.button_back).setOnClickListener(this)
    }

    private fun initializeView(acResult: AcResult) {
        (findViewById<View>(R.id.ac_date) as TextView).text = acResult.acDate
        (findViewById<View>(R.id.ac_time) as TextView).text = acResult.acTime
        val viewValue = findViewById<TextView>(R.id.ac_value)
        viewValue.text = acResult.acValue
        val value = acResult.acValue.substring(0, 5).toFloat()
        if (value >= 0.16f) {
            viewValue.setTextColor(ContextCompat.getColor(this, R.color.red))
        } else {
            viewValue.setTextColor(ContextCompat.getColor(this, R.color.green))
        }
        if (supportActionBar != null) supportActionBar!!.title = acResult.deviceName
        // ((TextView) findViewById(R.id.device_name)).setText(acResult.getDeviceName());
    }

    override fun onClick(v: View) {
        if (v.id == R.id.button_back) finish()
    }

    companion object {
        const val EXTRA_ACRESULT = "acresult"
    }
}